### PR TITLE
[RM Ch07] SBOM content added

### PR DESCRIPTION
**Pull request by karinesevilla**
Fixes issue #2793

### CODE REVIEW

**Review by walterkozlowski — APPROVED** [review]

File: doc/ref_model/chapters/chapter07.rst [hunk 1]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
+| **Version of the Component** | Identifier used by the supplier to specify a      |
+|                              | change in software from a previously identified   |
+|                              | version.                                          |
++------------------------------+---------------------------------------------------+
+| **Other Unique Identifiers** | Other identifiers that are used to identify a     |
+|                              | component, or serve as a look-up key for relevant |
+|                              | databases.                                        |
++------------------------------+---------------------------------------------------+
+| **Dependency Relationship**  | Characterizing the relationship that an upstream  |
+|                              | component X is included in software Y.            |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.rst:752]: Is it software or a component? I realise that software is also a comonent

File: doc/ref_model/chapters/chapter07.rst [hunk 2]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
+| **Author of SBOM Data**      | The name of the entity that creates the SBOM data |
+|                              | for this component.                               |
++------------------------------+---------------------------------------------------+
+| **Timestamp**                | Record of the date and time of the SBOM data      |
+|                              | assembly.                                         |
++------------------------------+---------------------------------------------------+
+**Table 7-2 **: SBOM Data Fields components, source `NTIA <https://www.ntia.doc.gov/files/ntia/publications/\
+sbom_minimum_elements_report.pdf>`_
+
+Refer to the document for more details on each data field. Examples of commonly used identifiers are provided.
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.rst:763]:
```suggestion
Refer to the NTIA SBOM document for more details on each data field. Examples of commonly used identifiers are provided.
```

File: doc/ref_model/chapters/chapter07.rst [hunk 3]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
+
+In order to use SBOMs efficiently and spread their adoption, information must be generated and shared in a standard 
+format. This format must be machine-readable to allow automation. Proprietary formats should not be used. 
+Multiple data formats exist covering baseline SBOM information. The three key formats, Software Package Data 
+eXchange (SPDX), CycloneDX, and Software Identification Tags (SWID tags) are interoperable for the core data fields 
+and use common data syntax representations.
+
+- `SPDX <https://spdx.dev/>`__ is an open-source machine-readable format developed under the umbrella of the Linux 
+  Foundation. 
+
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter07.rst:773]: can we remove the breaks to avoid confusion for the last paragraph in this sub-section,

**Comment by walterkozlowski** [comment]
@karinesevilla, can you please resolve correction suggestions from @pgoyal01? I will re-review after that.

File: doc/ref_model/chapters/chapter07.rst [hunk 4]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
-For both formats, SPDX and SWID Tags, tools are available.
+In order to ensure software security, it is crucial to identify the software components and their origins. The 
+Software Bill of Materials (SBOM), described by `US NTIA <https://www.ntia.gov/SBOM>`__ (National 
+Telecommunications and Information Administration), is an efficient and highly recommended tool to identify software 
+components. The SBOM is an inventory of software components and the relationships between them. NTIA describes how to 
+establish an SBOM and provides SBOM standard data formats. In case of vulnerability detected for a component, the 
+SBOM inventory is an effective means to identify the impacted component and provide remediation.
+
+A transparent software supply chain offers benefits for vulnerabilities remediation, but also for licensing management 
+and it provides assurance of the source and integrity of components. To achieve and take benefit of this transparency,
```
[the remark this answers is not in the document]
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst]: Thanks @pgoyal01, committed

File: doc/ref_model/chapters/chapter07.rst [hunk 5]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
++------------------------------+---------------------------------------------------+
+| **Version of the Component** | Identifier used by the supplier to specify a      |
+|                              | change in software from a previously identified   |
+|                              | version.                                          |
++------------------------------+---------------------------------------------------+
+| **Other Unique Identifiers** | Other identifiers that are used to identify a     |
+|                              | component, or serve as a look-up key for relevant |
+|                              | databases.                                        |
++------------------------------+---------------------------------------------------+
+| **Dependency Relationship**  | Characterizing the relationship that an upstream  |
```
[the remark this answers is not in the document]
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst:751]: It is the description of the NTIA document. "Upstream" in this definition must not be seen as open-source upstream code, it is more general. It represents  the initial component part of a piece of software

File: doc/ref_model/chapters/chapter07.rst (hunk 1 above)
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst:752]: The same as the previous answer; It is the exact definition of NTIA document.
A software is composed of multiple components, these components are software

File: doc/ref_model/chapters/chapter07.rst [hunk 6]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
+Software Bill of Materials (SBOM), described by `US NTIA <https://www.ntia.gov/SBOM>`__ (National 
+Telecommunications and Information Administration), is an efficient and highly recommended tool to identify software 
+components. The SBOM is an inventory of software components and the relationships between them. NTIA describes how to 
+establish an SBOM and provides SBOM standard data formats. In case of vulnerability detected for a component, the 
+SBOM inventory is an effective means to identify the impacted component and provide remediation.
+
+A transparent software supply chain offers benefits for vulnerabilities remediation, but also for licensing management 
+and it provides assurance of the source and integrity of components. To achieve and take benefit of this transparency, 
+a shared model must be supported by industry. This is the goal of the work performed by the USA department of commerce 
+and the National Telecommunications and Information administration (NTIA) and published, in July 2021, in the report
```
[the remark this answers is not in the document]
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst:725]: committed

File: doc/ref_model/chapters/chapter07.rst [hunk 7]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
-`SPDX <https://spdx.dev/>`__ is an open-source machine-readable format developed under the umbrella of the Linux Foundation. 
-The `SPDX specification 2.2 <https://spdx.dev/specifications/>`__ has been published as the standard ISO/IEC 5962:2021. It 
-provides a language for communicating the data, licenses, copyrights, and security information associated with software 
-components. With the SPDX specification 2.2, multiple file formats are available: YAML, JSON, RDF/XML, tag\:value flat text, 
-and xlss spreadsheets.
-
-`SWID Tags <https://nvd.nist.gov/products/swid>`__ is an international XML-based standard used by commercial software 
-publishers and has been published as the standard ISO/IEC 19770-2. The specification defines four types of SWID tags: primary, 
-patch, corpus, and supplemental to describe a software component.
+Software Bill of Materials (SBOM)
```
Comment (gkunz) [inline doc/ref_model/chapters/chapter07.rst:712]: Do we want to add pointers at the end of this section to other open source forums in which best practices for establishing, adopting and implementing SBOMs are being driven? I am thinking of OpenSSF and OpenChain here. The Anuket RAs and RIs will likely refer to SBOMs (RA) or use them (RI), but the best practices are defined elsewhere.

File: doc/ref_model/chapters/chapter07.rst [hunk 8]
```diff
@@ -708,33 +709,91 @@ Access to the repository must be granted by a dedicated authorization. The code
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-For both formats, SPDX and SWID Tags, tools are available.
+In order to ensure software security, it is crucial to identify the software components and their origins. The 
+Software Bill of Materials (SBOM), described by `US NTIA <https://www.ntia.gov/SBOM>`__ (National 
+Telecommunications and Information Administration), is an efficient and highly recommended tool to identify software 
+components. The SBOM is an inventory of software components and the relationships between them. NTIA describes how to 
+establish an SBOM and provides SBOM standard data formats. In case of vulnerability detected for a component, the 
+SBOM inventory is an effective means to identify the impacted component and provide remediation.
```
[the remark this answers is not in the document]
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst]: @gkunz Thanks! Change accepted

File: doc/ref_model/chapters/chapter07.rst (hunk 7 above)
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst:712]: @gkunz Yes, we can add pointers to other open source forums. We will have to enrich this section over time. I propose to keep it for Nile release.

**Comment by walterkozlowski** [comment]
@gkunz and @pgoyal01, can you please check whether your comments have been addressed, and approve, if you can.

File: doc/ref_model/chapters/chapter07.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (pgoyal01) [reply doc/ref_model/chapters/chapter07.rst:751]: @karinesevilla Thanks - can we add a note to that effect?

File: doc/ref_model/chapters/chapter07.rst (hunk 3 above)
Comment (pgoyal01) [reply doc/ref_model/chapters/chapter07.rst:773]: @karinesevilla No I was thinking of making each of SPDX, CycloneDX, etc. as bullets like in the following (with line breaks for line length limits):
- `SPDX <https://spdx.dev/>`__ is an open-source machine-readable format developed under the umbrella of the Linux Foundation.  The `SPDX specification 2.2 <https://spdx.dev/specifications/>`__ has been published as the standard  ISO/IEC 5962:2021. It provides a language for communicating the data, licenses, copyrights, and security information associated with software components. With the SPDX specification 2.2, multiple file formats are available: YAML,  JSON, RDF/XML, tag\:value flat text, and xls spreadsheets.
- CYcloneDX ...


It is OK if you want to leave as is.

File: doc/ref_model/chapters/chapter07.rst (hunk 7 above)
Comment (gkunz) [reply doc/ref_model/chapters/chapter07.rst:712]: Fine with me.

**Review by gkunz — APPROVED** [review]

**Review by pgoyal01 — APPROVED** [review]

File: doc/ref_model/chapters/chapter07.rst (hunk 3 above)
Comment (karinesevilla) [reply doc/ref_model/chapters/chapter07.rst:773]: @pgoyal01 Okay I see, I will do it during proofreading

**Pull request merged**